omnisend/includes/SDK/V1/class-event.php

```diff
@@ -0,0 +1,181 @@
+<?php
+/**
+ * Omnisend Client
+ *
+ * @package OmnisendClient
+ */
+
+namespace Omnisend\SDK\V1;
+
+use Omnisend\SDK\V1\Contact;
+use WP_Error;
+
+defined( 'ABSPATH' ) || die( 'no direct access' );
+
+/**
+ * Omnisend Event class. It's should be used with Omnisend Client.
+ *
+ */
+class Event {
+
+
+	private $contact          = null;
+	private $event_name       = null;
+	private $event_time       = null;
+	private $event_version    = null;
+	private $origin           = null;
+	private array $properties = array();
+
+	/**
+	 * Validate event properties.
+	 *
+	 * It ensures that all required properties are set
+	 *
+	 * @return WP_Error
+	 */
+	public function validate(): WP_Error {
+
+		$error = new WP_Error();
+		if ( $contact instanceof Contact ) {
+			$error->merge_from( $contact->validate() );
+		}
+
+		if ( $this->$event_name == null ) {
+			$error->add( 'event_name', 'Is required.' );
+		}
+
+		if ( $this->$event_name != null && ! is_string( $this->$event_name ) ) {
+			$error->add( $event_name, 'Not a string.' );
+		}
+
+		if ( $this->$event_version != null && ! is_string( $this->$event_version ) ) {
+			$error->add( $event_version, 'Not a string.' );
+		}
+
+		if ( $this->$origin != null && ! is_string( $this->$origin ) ) {
```

Comment:
Origin is also mandatory, please add check same as for event name.

Reply:
I dont think it is required, in API documentation we have

```
origin string
It should be based on the source of event API. If you are building platform or app integration it should have dedicated origin. Otherwise it should be "api".
```

Reply:
Maybe documentation is not clear enough. Just checked, it is required.

Reply:
![image](https://github.com/omnisend/wp-omnisend/assets/98449470/8119f766-8343-4a59-ba73-dda7e47eba65)

Reply:
lovely... Will add validation

Reply:
Oh, ok, it says "Otherwise it should be api", this means users should send value api

Reply:
Yes, please pass "api" if not provided ;)

Reply:
This is why we build SDK to reduce pain from all these details from our SDK user.

Reply:
ok